buildSrc/src/test/java/quilt/internal/TasksTest.java

```diff
@@ -0,0 +1,326 @@
+        assertEquals("right", insertedMethod3.getArg(-1, 2, null).getName("named"));
+    }
+
+    @Test
+    @Order(5)
+    public void testMergeTinyV2() throws Exception {
+        Path output = outputsDir.resolve("merged-v2.tiny");
+        mergeTinyV2Output = output;
+        MergeTinyV2Task.mergeMappings(insertAutoGeneratedMappingsOutput, invertedPerVersionMappings, output);
+
+        MappingTree tree = TestUtil.readTinyV2(output);
+        testMergedTree(tree, PER_VERSION_NAMESPACE);
+        // com/example/u/C_002 f_004 FIRST
+        assertEquals("FIRST", tree.getField("b", "a", "Lb;").getName("named"));
+        // com/example/u/C_003$C_004 f_011 NORMAL
+        assertEquals("NORMAL", tree.getField("c$a", "a", "Lc$a;").getName("named"));
+    }
+
+    private void mapNamedJar() {
+        if (namedJar != null) return;
+        namedJar = testProjectDir.resolve("test-named.jar");
+        JarRemapper.mapJar(namedJar, perVersionMappingsJar, mergeTinyV2Output, new Path[0], PER_VERSION_NAMESPACE, "named", null);
+    }
+
+    @Test
+    @Order(6)
+    public void testCombineUnpickDefinitions() throws Exception {
+        Path output = outputsDir.resolve("combined-definitions.unpick");
+        combineUnpickDefinitionsOutput = output;
+        Collection<File> input = List.of(unpickDefinitions.toFile().listFiles());
+        CombineUnpickDefinitionsTask.combineUnpickDefinitions(input, output);
+
+        mapNamedJar();
+        UnpickFile file = TestUtil.readUnpickFile(output, namedJar);
+        assertTrue(file.containsGroup("e_flags"));
+        assertTrue(file.isFlagGroup("e_flags"));
+        assertEquals("e_flags", file.getParameterConstantGroup("com/example/EClass", "m1", "(I)V", 0));
+
+        assertTrue(file.containsGroup("f_flags"));
+        assertFalse(file.isSimpleConstantGroup("f_flags"));
+        assertEquals("f_flags", file.getParameterConstantGroup("com/example/FEnum", "<init>", "(Ljava/lang/String;II)V", 0));
+
+        assertTrue(file.containsGroup("g_type"));
+        assertTrue(file.isSimpleConstantGroup("g_type"));
+        assertEquals("g_type", file.getParameterConstantGroup("com/example/GClass", "fromType", "(I)Ljava/lang/String;", 0));
+        assertEquals("g_type", file.getReturnConstantGroup("com/example/GClass", "toType", "(Ljava/lang/String;)I"));
+    }
+
+    @Test
+    @Order(7)
+    public void testBuildTinyWithEnum() throws Exception {
+        Path output = outputsDir.resolve("buildTinyWithEnum.tiny");
+        buildTinyWithEnumOutput = output;
+        AddProposedMappingsTask.addProposedMappings(mergeTinyOutput, output, OBF_JAR, profilePath);
+
+        MappingTree inputTree = TestUtil.readTinyV2(mergeTinyOutput);
+        MappingTree tree = TestUtil.readTinyV2(output);
+        assertEquals(inputTree.getSrcNamespace(), tree.getSrcNamespace());
+        assertEquals(inputTree.getDstNamespaces(), tree.getDstNamespaces());
+
+        // TODO: Fix buildTinyWithEnum
```

Comment:
what's the status of this todo?

Reply:
I'll fix it in another pr, this one is already quite big, and I still have to figure out how to fix it